File: scrape.py
from playwright.sync_api import sync_playwright
import re

seeds = list(range(32, 42))

total_sum = 0

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page()

    for seed in seeds:
        url = f"https://sanand0.github.io/tdsdata/playwright-table/?seed={seed}"
        page.goto(url)
        page.wait_for_load_state("networkidle")

        tables = page.query_selector_all("table")
        for table in tables:
            text = table.inner_text()
            numbers = re.findall(r"-?\d[\d,]*\.?\d*", text)
            numbers = [float(n.replace(",", "")) for n in numbers]
            total_sum += sum(numbers)

    browser.close()

print("FINAL TOTAL:", total_sum)

File: .github/actions/scripts/scrape.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  let grandTotal = 0;

  const urls = [
    'https://sanand0.github.io/tdsdata/js_table/?seed=32',
    'https://sanand0.github.io/tdsdata/js_table/?seed=33',
    'https://sanand0.github.io/tdsdata/js_table/?seed=34',
    'https://sanand0.github.io/tdsdata/js_table/?seed=35',
    'https://sanand0.github.io/tdsdata/js_table/?seed=36',
    'https://sanand0.github.io/tdsdata/js_table/?seed=37',
    'https://sanand0.github.io/tdsdata/js_table/?seed=38',
    'https://sanand0.github.io/tdsdata/js_table/?seed=39',
    'https://sanand0.github.io/tdsdata/js_table/?seed=40',
    'https://sanand0.github.io/tdsdata/js_table/?seed=41'
  ];

  for (const url of urls) {
    const page = await browser.newPage();

    console.log(`Visiting: ${url}`);
    await page.goto(url, { waitUntil: 'networkidle' });

    const numbers = await page.$$eval('table td, table th', elements =>
      elements.flatMap(el => {
        const text = el.textContent.trim();
        const num = parseFloat(text.replace(/[^\d.-]/g, ''));
        return isNaN(num) ? [] : [num];
      })
    );

    const pageSum = numbers.reduce((a, b) => a + b, 0);
    grandTotal += pageSum;

    console.log(`Sum for ${url}: ${pageSum}`);
    await page.close();
  }

  console.log(`GRAND TOTAL SUM OF ALL TABLES: ${grandTotal}`);
  await browser.close();
})();
